File: 04.Simulation模拟/leetcode_1732_largestAltitude.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int res = 0, cur = 0;
        for (int i : gain){
            cur += i;
            res = max(res, cur);
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> gain = {-5,1,5,0,-7};
    cout << s.largestAltitude(gain) << endl;
}

File: 04.Simulation模拟/leetcode_1732_largestAltitude.py
from typing import List

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        res = 0
        cur = 0
        for i in gain:
            cur += i
            res = max(res, cur)
        return res

if __name__ == "__main__":
    s = Solution()
    gain = [-5,1,5,0,-7]
    print(s.largestAltitude(gain))
